addons/purchase_stock/models/purchase.py

```diff
@@ -475,7 +475,7 @@ def _prepare_stock_move_vals(self, picking, price_unit, product_uom_qty, product
         return {
             # truncate to 2000 to avoid triggering index limit error
             # TODO: remove index in master?
-            'name': (self.name or '')[:2000],
+            'name': (self.product_id.display_name or '')[:2000],
```

Comment:
I think this is a risky fix. `display_name` is dependent on:

- context values
https://github.com/odoo/odoo/blob/85a634c7527f20b56b15c2da90a37dfd637da63b/addons/product/models/product.py#L457
https://github.com/odoo/odoo/blob/85a634c7527f20b56b15c2da90a37dfd637da63b/addons/product/models/product.py#L492
- current company
https://github.com/odoo/odoo/blob/85a634c7527f20b56b15c2da90a37dfd637da63b/addons/product/models/product.py#L462

and maybe also the user's language setting? (not sure, didn't check).  Using this will also potentially add computation time :thinking: 

Is it not possible to do something like change the line you mentioned:

`<t t-set="has_kits" t-value="o.move_line_ids.filtered(lambda l: l.move_id.bom_line_id and l.move_id.name in (l.move_id.bom_line_id.bom_id.product_tmpl_id.display_name, l.move_id.bom_line_id.bom_id.product_id.display_name))"/>`

to

`<t t-set="has_kits" t-value="o.move_line_ids.filtered(lambda l: l.move_id.bom_line_id and l.move_id.product_id.display_name in (l.move_id.bom_line_id.bom_id.product_tmpl_id.display_name, l.move_id.bom_line_id.bom_id.product_id.display_name))"/>`

? (i.e. match against the `l.move.product_id.display_name` instead of `move.name` I didn't test if it works)

Reply:
Hello @ticodoo,
Thanks for the review, I agree it's risky, but it's more likely to work than using the sale order line description as the name of the move. But, indeed, if the customer changes the name of the product after the SO confirmation, it will be problematic, because the move name would not be updated, therefore not compatible. For the computation time, it only adds a short time, it is calculated only once, so it should not impact the performance.

Using `l.move.product_id.display_name instead of `move.name` will not work because the move.product_id is the component and not the kit product, therefore we will not be able to check if it is linked to the BOM.

What do you think ?

Reply:
OK, I thought about it some more. Since I'm not sure if `move.name` will be used elsewhere, can we do something like `... and any(l.move_id.name.split('\n') in (l.move_id.bom_line_id.bom_id.product_tmpl_id.display_name, l.move_id.bom_line_id.bom_id.product_id.display_name))`? I realized that both purchase/sale descriptions are automatically written as `product display name + \n + custom description` so we can possibly still check without changing the underlying data in the log run. Of course this fails if the user changes the description to not include the product name at all.... What do you think?